packages/x-date-pickers/src/YearCalendar/YearCalendar.tsx

```diff
@@ -2,13 +2,13 @@ import * as React from 'react';
 import PropTypes from 'prop-types';
 import clsx from 'clsx';
 import { useRtl } from '@mui/system/RtlProvider';
-import { styled, useThemeProps } from '@mui/material/styles';
```

Comment:
To keep the static-extraction intact we should avoid adding new imports from `@mui/material/styles` directly.
WDYT about creating an ESLint rule for that? 🤔 
However, we'd have to think of what to do with type imports... Maybe `import type` could be skipped by an ESLint rule? 🤔

Reply:
@brijeshb42 is this something that is planned on the core side?
I guess it could benefit all MUI teams trying to migrate and to avoid re-using direct imports by mistake.

I don't think we can totally bloke `@mui/material/styles` since we want to use `useTheme`

Reply:
Makes sense. We'd need too many ignores for those. 🙈 

> I don't think we can totally bloke `@mui/material/styles` since we want to use `useTheme`

Which ideally would be eventually replaced with the `@mui/system` import if there were no issues with theme merging/scoping. 🤔 🤞

Reply:
Yes, not sure when this would be though